compiler/noirc_frontend/src/hir/resolution/errors.rs

```diff
@@ -386,6 +388,11 @@ impl<'a> From<&'a ResolverError> for Diagnostic {
                 diag.add_note("The `#[fold]` attribute specifies whether a constrained function should be treated as a separate circuit rather than inlined into the program entry point".to_owned());
                 diag
             }
+            ResolverError::ArrayLengthInterpreter { error, span } => Diagnostic::simple_error(
+                "Interpreting array length failed".into(),
+                format!("with error: {}", Diagnostic::from(error)),
+                *span,
```

Comment:
Same here - maybe we should just display the error directly?

Reply:
Agreed on this. Ideally we'd surface the original error directly or at least just delegate to it.